integrations/langchain/src/databricks_langchain/vector_search_retriever_tool.py

```diff
@@ -0,0 +1,81 @@
+from typing import Any, Dict, List, Optional
+
+from pydantic import BaseModel, Field, model_validator, PrivateAttr
+
+from databricks_langchain import DatabricksVectorSearch
+from databricks_langchain.utils import IndexDetails
+from langchain_core.embeddings import Embeddings
+from langchain_core.tools import BaseTool
+
+
+class VectorSearchRetrieverTool(BaseTool):
+    """
+    A utility class to create a vector search-based retrieval tool for querying indexed embeddings.
+    This class integrates with a Databricks Vector Search and provides a convenient interface
+    for building a retriever tool for agents.
+    """
+
+    index_name: str = Field(..., description="The name of the index to use, format: 'catalog.schema.index'.")
+    num_results: int = Field(10, description="The number of results to return.")
+    columns: Optional[List[str]] = Field(None, description="Columns to return when doing the search.")
+    filters: Optional[Dict[str, Any]] = Field(None, description="Filters to apply to the search.")
+    query_type: str = Field("ANN", description="The type of query to run.")
+    tool_name: Optional[str] = Field(None, description="The name of the retrieval tool.")
+    tool_description: Optional[str] = Field(None, description="A description of the tool.")
+    # TODO: Confirm if we can add these two to the API to support direct-access indexes or a delta-sync indexes with self-managed embeddings,
+    text_column: Optional[str] = Field(None, description="If using a direct-access index or delta-sync index, specify the text column.")
+    embedding: Optional[Embeddings] = Field(None, description="Embedding model for self-managed embeddings.")
+    # TODO: Confirm if we can add this endpoint field
+    endpoint: Optional[str] = Field(None, description="Endpoint for DatabricksVectorSearch.")
+
+    # The BaseTool class requires 'name' and 'description' fields which we will populate in validate_tool_inputs()
+    name: str = Field(default="", description="The name of the tool")
+    description: str = Field(default="", description="The description of the tool")
+
+    _vector_store: DatabricksVectorSearch = PrivateAttr()
+
+    @model_validator(mode='after')
+    def validate_tool_inputs(self):
+        # Construct the vector store using provided params
+        kwargs = {
+            "index_name": self.index_name,
+            "endpoint": self.endpoint,
+            "embedding": self.embedding,
+            "text_column": self.text_column,
+            "columns": self.columns,
+        }
+        dbvs = DatabricksVectorSearch(**kwargs)
+        self._vector_store = dbvs
+
+        def get_tool_description():
+            default_tool_description = "A vector search-based retrieval tool for querying indexed embeddings."
+            index_details = IndexDetails(dbvs.index)
+            if index_details.is_delta_sync_index():
```

Comment:
direct access indexes don't have an associated source table so we'll just use the default tool description.

Reply:
Curious what the existing `langchain-databricks` DatabricksVectorSearch.as_retriever(...).as_tool(...) ends up generating as the tool description

Reply:
One way to tell would be to use it as a tool with payload logging enabled & see what the `tools` argument to the LLM API in model serving looks like

Reply:
This generally looks reasonable, just curious if we can keep it in sync with the existing behavior/default

Reply:
<img width="1016" alt="image" src="https://github.com/user-attachments/assets/aa237f4a-98a8-4c8b-a4dd-f5654944fe34" />
Tested it out in a notebook. The default seems to be extremely basic and lacking in content. Does this answer your question?

Reply:
Lol yep makes sense, the updated version in this PR is definitely better